### PR TITLE
Various cleanups to avoid compile errors and warnings building Chunreal for macOS

### PR DESCRIPTION
**Pull request by max-jardetzky**

### CODE REVIEW

File: src/core/chuck_dl.cpp [hunk 1]
```diff
@@ -1949,11 +1949,11 @@ remove_all_shreds(ck_remove_all_shreds)
...
 Chuck_DL_Api::ObjectApi::ObjectApi() :
 get_type(ck_get_type),
-create(ck_create_with_shred),
-create_without_shred(ck_create_without_shred),
 add_ref(ck_add_ref),
 release(ck_release),
 refcount(ck_refcount),
+create(ck_create_with_shred),
+create_without_shred(ck_create_without_shred),
 create_string(ck_create_string),
```
Comment (max-jardetzky) [inline src/core/chuck_dl.cpp:1957]: reordered constructor arguments relative to the member declaration order

File: src/core/chuck_emit.cpp [hunk 2]
```diff
@@ -3636,8 +3636,8 @@ t_CKBOOL emit_engine_emit_exp_primary( Chuck_Emitter * emit, a_Exp_Primary exp )
         }
         else if( exp->var == insert_symbol( "pi" ) )
         {
-            double pi = 3.14159265358979323846;
-            emit->append( new Chuck_Instr_Reg_Push_Imm2( pi ) );
+            double ckPi = 3.14159265358979323846;
+            emit->append( new Chuck_Instr_Reg_Push_Imm2( ckPi ) );
         }
```
Comment (max-jardetzky) [inline src/core/chuck_emit.cpp:3641]: renamed `pi` due to global namespace pollution in the UE environment on macOS (some system framework defined `pi` already)

File: src/core/chuck_type.cpp [hunk 3]
```diff
@@ -8205,16 +8205,16 @@ static t_CKBOOL g_escape_ready = FALSE;
...
+    g_escape[(t_CKUINT)'\\'] = '\\';
+    g_escape[(t_CKUINT)'a'] = (char)7; // audible bell
+    g_escape[(t_CKUINT)'b'] = (char)8; // back space
+    g_escape[(t_CKUINT)'f'] = (char)12; // form feed
+    g_escape[(t_CKUINT)'n'] = (char)10; // new line
+    g_escape[(t_CKUINT)'r'] = (char)13; // carriage return
+    g_escape[(t_CKUINT)'t'] = (char)9; // horizontal tab
+    g_escape[(t_CKUINT)'v'] = (char)11; // vertical tab
 
     // done
```
Comment (max-jardetzky) [inline src/core/chuck_type.cpp:8219]: some implicit conversion warning, made the conversion explicit

File: src/core/chuck_type.h [hunk 4]
```diff
@@ -1018,11 +1018,11 @@ struct Chuck_Type : public Chuck_Object
...
     {
+        // the callback
+        f_callback_on_instantiate callback;
         // whether to auto-set shred origin at instantiation;
         // see t_CKBOOL initialize_object( ... )
         t_CKBOOL shouldSetShredOrigin;
-        // the callback
-        f_callback_on_instantiate callback;
         // constructor
         CallbackOnInstantiate( f_callback_on_instantiate cb = NULL, t_CKBOOL setShredOrigin = FALSE )
```
Comment (max-jardetzky) [inline src/core/chuck_type.h:1027]: reordered members relative to their usage order in the constructor

File: src/core/ugen_stk.cpp [hunk 5]
```diff
@@ -8486,7 +8486,7 @@ void Echo :: set( MY_FLOAT max )
     MY_FLOAT delay = delayLine ? delayLine->getDelay() : length>>1;
     if( delayLine ) delete delayLine;
     if( delay >= max ) delay = max;
-    delayLine = new Delay(length>>1, length);
+    delayLine = new class Delay(length>>1, length);
     this->clear();
```
Comment (max-jardetzky) [inline src/core/ugen_stk.cpp:8490]: global namespace pollution fix: `Delay` is some global function in UE on macOS, a C++ language feature allows us to disambiguate with the `class` specifier before usages of the class

File: src/core/ugen_stk.h [hunk 6]
```diff
@@ -821,10 +822,10 @@ class DelayL : public Delay
   /*!
     This method is valid only for delay settings greater than zero!
    */
-  MY_FLOAT nextOut(void);
+  virtual MY_FLOAT nextOut(void);
```
Comment (max-jardetzky) [inline src/core/ugen_stk.h:826]: this class of warnings came from implicit virtual function shadowing or overriding, the fix seems to be making overridden methods virtual and adding `using` declarations before overloaded methods

**Review by gewang — APPROVED** [review]
excellent; thank you!

**Pull request merged**